src/player.cpp

```diff
@@ -3865,14 +3865,44 @@ void Player::addUnjustifiedDead(const Player* attacked)
 
 	sendTextMessage(MESSAGE_EVENT_ADVANCE, "Warning! The murder of " + attacked->getName() + " was not justified.");
 
-	skullTicks += g_config.getNumber(ConfigManager::FRAG_TIME);
+	time_t now = time(nullptr), today = (now - 84600), week = (now - (7 * 84600));
+	std::vector<time_t> killsList;
+	killsList.push_back(now); // add current kill to list
+	IOLoginData::getUnjustifiedDates(name, killsList, now); // add kills from last month to list
+	int32_t todayKills = 0, weekKills = 0, monthKills = killsList.size();
 
-	if (getSkull() != SKULL_BLACK) {
-		if (g_config.getNumber(ConfigManager::KILLS_TO_BLACK) != 0 && skullTicks > (g_config.getNumber(ConfigManager::KILLS_TO_BLACK) - 1) * static_cast<int64_t>(g_config.getNumber(ConfigManager::FRAG_TIME))) {
-			setSkull(SKULL_BLACK);
-		} else if (getSkull() != SKULL_RED && g_config.getNumber(ConfigManager::KILLS_TO_RED) != 0 && skullTicks > (g_config.getNumber(ConfigManager::KILLS_TO_RED) - 1) * static_cast<int64_t>(g_config.getNumber(ConfigManager::FRAG_TIME))) {
+	for (const auto& it : killsList) {
+		if (it > week) {
+			weekKills++;
+		}
+
+		if (it > today) {
+			todayKills++;
```

Comment:
```suggestion
			++todayKills;
```

Reply:
May I ask what's the difference, when both variables are used outside of a loop?

Reply:
Semantics. Only use `x++` if you are capturing the previous value, i.e. `y = x++`, since they don't have the same meaning.

If by not outside a loop you mean the `x++` in `for (...; ...; x++)`, note that even in that case the expressions are similar (you could use `++x` without change in behavior)